admin/src/main/java/org/bahmni/module/admin/csv/persister/LabResultPersister.java

```diff
@@ -83,17 +79,16 @@ public Messages persist(LabResultsRow labResultsRow) {
             encounter.addProvider(encounterService.getEncounterRoleByUuid(EncounterRole.UNKNOWN_ENCOUNTER_ROLE_UUID), getProvider());
             HashSet<Obs> resultObservations = new HashSet<>();
             for (LabResultRow labResultRow : labResultsRow.getTestResults()) {
-                labResultRow.getResult();
                 org.openmrs.Concept concept = conceptService.getConceptByName(labResultRow.getTest());
-                if(concept.isNumeric()){
-                    ConceptNumeric cn = (ConceptNumeric)concept;
-                    if(!cn.isAllowDecimal() && labResultRow.getResult().contains(".")){
-                        throw new APIException("Decimal is not allowed for "+ cn.getName() +" concept");
+                if (concept.isNumeric()) {
+                    ConceptNumeric cn = (ConceptNumeric) concept;
+                    if (!cn.isAllowDecimal() && labResultRow.getResult().contains(".")) {
+                        throw new APIException("Decimal is not allowed for " + cn.getName() + " concept");
                     }
                 }
                 Order testOrder = getTestOrder(patient, labResultRow, labResultsRow.getTestDate());
                 encounter.addOrder(testOrder);
-                resultObservations.add(getResultObs(labResultRow, testOrder));
+                resultObservations.add(getResultObs(labResultRow, testOrder, concept));
```

Comment:
in the getResultObs() method, the variable 'labResultRow' is used only to get the string value of the result. Why not just pass a 'resultString' variable?

Reply:
Hi @angshu,
`labResultRow` row was already there. I just added a new parameter to `getResultObs`. If you want I can refactor it.

Reply:
I think you should.